Dockerfile

```diff
@@ -27,14 +27,16 @@ RUN yum install -y \
 # configure java runtime
 ENV JAVA_HOME=/opt/java \
   JAVA_VERSION_MAJOR=8 \
-  JAVA_VERSION_MINOR=102 \
-  JAVA_VERSION_BUILD=14
+  JAVA_VERSION_MINOR=112 \
+  JAVA_VERSION_BUILD=15
 
 # configure nexus runtime
+ENV SONATYPE_DIR=/opt/sonatype
 ENV NEXUS_VERSION=3.1.0-04 \
-  NEXUS_HOME=/opt/sonatype/nexus \
+  NEXUS_HOME=${SONATYPE_DIR}/nexus \
   NEXUS_DATA=/nexus-data \
-  NEXUS_CONTEXT=''
+  NEXUS_CONTEXT='' \
+  SONATYPE_WORK=${SONATYPE_DIR}/sonatype-work
```

Comment:
I believe the idea is to leave KARAF_DATA configured as `../sonatype-work/nexus3` and instead symlink `${NEXUS_DATA}` as `${SONATYPE_WORK}/nexus3` - rather than change KARAF_DATA which involves updating several entries in `nexus.vmoptions` (because that file doesn't support property interpolation).

If install4j ever added property interpolation to `nexus.vmoptions` that would avoid the need to update entries to reflect the new property value, but for now we either need to edit it wholesale or take Brad's symlink approach.